src/librustdoc/test.rs

```diff
@@ -76,7 +76,16 @@ pub fn run(options: Options) -> i32 {
         lint_caps: Default::default(),
     };
 
-    let mut test_args = options.test_args.clone();
+    let mut test_args = options.test_args
+        .iter()
+        .filter_map(|arg| {
```

Comment:
`.filter(..).cloned()` seems cleaner here.

Reply:
I continue to be deeply confused why we're altering test_args here.

test_args is the array of arguments that's passed to test::test_main -- that's way too late for `cfg` to matter, I believe, and test_main certainly is not a compiler.

Can you at least check manually that this works? A comment that explains how this works (e.g., why is this applying to cfg, despite not mentioning cfg at all?) would also be good.

Reply:
I just thought about it when you said something was wrong in this implementation the other day: it's incorrect since I don't check the `cfg`. However, `test_args` are passed to each code sample rustdoc compiles and run. Therefore, it seems accurate to me to have this check here. However I need to fix it.

Reply:
But... they're not? These are not the arguments to rustc. These are the arguments to the test 'binary'. I would expect you to _never_ find a `--cfg` in these arguments; they're not (AFAICT) used at all for compilation. Right now the code is removing a test _filter_ for rustdoc -- not a cfg gate or anything like that.

Reply:
Then I completely misunderstood this part of the code... I'll take a step back and check what I missed...